sktime/utils/validation/forecasting.py

```diff
@@ -341,3 +359,87 @@ def check_scoring(scoring):
         raise TypeError(f"`scoring` must inherit from `{allowed_base_class.__name__}`")
 
     return scoring
+
+
+def check_y_true_pred(y_true, y_pred):
+    """Validates the y_true and y_pred input data to forecasting performance
+    metrics.
+
+    Parameters
+    ----------
+    y_true : pandas Series, pandas DataFrame or NumPy ndarray of
+            shape (fh,) or (fh, n_columns) where fh is the forecasting horizon
+        Ground truth (correct) target values.
+
+    y_pred : pandas Series, pandas DataFrame or NumPy ndarray of
+            shape (fh,) or (fh, n_columns) where fh is the forecasting horizon
+        Estimated target values.
+
+    Returns
+    -------
+    actual : NumPy ndarray of shape (fh,) or (fh, n_columns)
+        The actual values of the input series in y_true
+
+    forecast : NumPy ndarray of shape (fh,) or (fh, n_columns)
+        The forecasts in the input y_pred
+
+    Raises
+    ------
+    TypeError
+        The type of y_true and y_pred are not equal
+    ValueError
+        Equal dimension required for y_true and y_pred
+    ValueError
+        Equal number of observations required of y_true and y_pred
+    ValueError
+        Equal numnber of columns required for y_true and y_pred
+    """
+    # Includes pd.Series, pd.DataFrame, np.ndarray
+    y_true = check_y(y_true, enforce_univariate=False, allow_numpy=True)
```

Comment:
I'm happy to accept numpy arrays for y in the performance metrics

Reply:
Great. I think that can be useful in some cases, particularly if someone wants to convert the functions into a scikit-learn scorer (I believe that based on how I've coed them they should technically work with non-timeseries data even though we describe them in a timeseries context in the docs).

Reply:
Yes, I agree, see my above comment!